tests/foreman/cli/test_classparameters.py

```diff
@@ -0,0 +1,827 @@
+        """Override the Default Parameter value.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Set the new valid Default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter Value overridden with new value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_override_uncheck(self):
+        """Override the Default Parameter value - override Unchecked.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Don't override the parameter.
+        2.  Set the new valid Default Value.
+        3.  Attempt to submit the changes.
+
+        @assert: Parameter value not allowed/disabled to override.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_puppet_default(self):
+        """On Override, Set Puppet Default Value.
+
+        @feature: Class Parameters - Puppet Default
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Set puppet default value to 'Use Puppet Default'.
+        3.  Submit the changes.
+
+        @assert: Puppet Default Value applied on parameter.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_parameter_type(self):
+        """Positive Parameter Update for parameter types - Valid Value.
+
+        Types - string, boolean, integer, real, array, hash, yaml, json
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Update the Key Type.
+        3.  Provide a 'valid' default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter Updated with a new type successfully.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_update_parameter_type(self):
+        """Negative Parameter Update for parameter types - Invalid Value.
+
+        Types - string, boolean, integer, real, array, hash, yaml, json
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Update the Key Type.
+        3.  Enter an 'Invalid' default Value.
+        3.  Submit the changes.
+
+        @assert:
+
+        1.  Parameter not updated with string type for invalid value.
+        2.  Error raised for invalid default value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_validate_puppet_default_value(self):
+        """Validation doesn't works on puppet default value.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Set puppet default value to 'Use Puppet Default'.
+        3.  Validate this value under section 'Optional Input Validator'.
```

Comment:
Yes, As asset says it should not work with puppet default value. The intention is to 'NOT' raise any error for puppet default value.